cupyx/fallback_mode/utils.py

```diff
@@ -33,11 +42,115 @@ def _call_numpy(func, args, kwargs):
     Returns:
         Result after calling func.
     """
+    args = _convert_ndarrays(args)
 
     numpy_args, numpy_kwargs = data_transfer._get_numpy_args(args, kwargs)
 
     numpy_res = func(*numpy_args, **numpy_kwargs)
 
     cupy_res = data_transfer._get_cupy_result(numpy_res)
 
-    return cupy_res
+    return _get_fallback_ndarray(cupy_res)
+
+
+def _convert_ndarrays(args):
+    return tuple(
+        [i._array if isinstance(i, ndarray) else i for i in args])
+
+
+def _get_fallback_ndarray(res):
+
+    if isinstance(res, cp.ndarray):
+        return ndarray(res)
+
+    return res
+
+
+def make_method(name):
+    def method(self, *args, **kwargs):
+
+        args = _convert_ndarrays(args)
+
+        cupy_method = getattr(cp.ndarray, name)
+
+        res = cupy_method(self._array, *args, **kwargs)
+
+        return _get_fallback_ndarray(res)
+
+    return method
+
+
+def _create_magic_methods():
+
+    _magic_methods = [
+
+        # Comparison operators:
+        '__richcmp__',
+
+        # Truth value of an array (bool):
+        '__nonzero__',
+
+        # Unary operations:
+        '__neg__', '__pos__', '__abs__', '__invert__',
+
+        # Arithmetic:
+        '__add__', '__sub__', '__mul__', '__matmul__', '__div__',
+        '__truediv__', '__floordiv__', '__mod__', '__divmod__', '__pow__',
+        '__lshift__', '__rshift__', '__and__', '__or__', '__xor__',
+
+        # Arithmetic, in-place:
+        '__iadd__', '__isub__', '__imul__', '__idiv__', '__itruediv__',
+        '__ifloordiv__', '__imod__', '__ipow__', '__ilshift__', '__irshift__',
+        '__iand__', '__ior__', '__ixor__',
+
+        # For standard library functions:
+        '__copy__', '__deepcopy__', '__reduce__',
+
+        # Container customization:
+        '__iter__', '__len__', '__getitem__', '__setitem__',
+
+        # Conversion:
+        '__int__', '__float__', '__complex__', '__oct__', '__hex__',
+
+        # String representations:
+        '__repr__', '__str__'
+    ]
+
+    for method in _magic_methods:
+        setattr(ndarray, method, make_method(method))
+
+
+class ndarray:
```

Comment:
This class should be defined in `numpy.py`

Reply:
There is no `numpy.py`. I defined `ndarray` class in `ndarray.py`, but that lead to relative imports between `utils.py` and `ndarray.py` beacause of `_get_fallback_ndarray` method.
Though we can try to pass `ndarray` through `_get_fallback_ndarray`. So, then should I define it in separate file?

Reply:
Sorry, I was mistaken. I want the following code to work.
```
from cupyx.fallbac_mode import numpy
numpy.ndaray((3, 4))  # Returns ndarray with (3, 4) shape
```

Reply:
This code already works.
Could you refer this:
https://github.com/cupy/cupy/pull/2272#discussion_r297487288